src/ray/protobuf/gcs.proto

```diff
@@ -161,6 +161,9 @@ message ActorTableData {
   // The actor's class name. This is necessary because the task spec's lifetime
   // is shorter than the ActorTableData.
   string class_name = 23;
+  // Only applies when state=DEAD. Indicates that this actor is marked
+  // as DEAD due to runtime environment setup failure.
+  bool runtime_env_setup_failed = 24;
```

Comment:
Instead of having this bool, can we create a new message like `message ActorDeadCause`? And within the dead cause, we can use `runtime_env_setup_failed ` as an enum. This will help us adding more observability into actor death in the future (we can add more dead states other than `runtime_env_setup_failed ` this way)

Reply:
I agree with you, I'll add one.